### ISSUE TITLE
feature selection: remove duplicated features

### ISSUE THREAD

**Issue by solegalli** [solegalli/feature_engine#114]
```
# check for duplicated features in the training set

# create and empty dictionary, where we will store 
# the duplicate groups
duplicated_feat_pairs = {}

# create an empty list to collect features
# that were found to be duplicated
_duplicated_feat = []


# iterate over every feature in our dataset:
for i in range(0, len(X_train.columns)):
    
    # this bit helps me understand where the loop is at:
    if i % 10 == 0:  
        print(i)
    
    # choose 1 feature:
    feat_1 = X_train.columns[i]
    
    # check if this feature has already been identified
    # as a duplicate of another one. If it was, it should be stored in
    # our _duplicated_feat list.
    
    # If this feature was already identified as a duplicate, we skip it, if
    # it has not yet been identified as a duplicate, then we proceed:
    if feat_1 not in _duplicated_feat:
    
        # create an empty list as an entry for this feature in the dictionary:
        duplicated_feat_pairs[feat_1] = []

        # now, iterate over the remaining features of the dataset:
        for feat_2 in X_train.columns[i + 1:]:

            # check if this second feature is identical to the first one
            if X_train[feat_1].equals(X_train[feat_2]):

                # if it is identical, append it to the list in the dictionary
                duplicated_feat_pairs[feat_1].append(feat_2)
                
                # and appended to our monitor list for duplicated variables
                _duplicated_feat.append(feat_2)
                
                # done! and run!
```

```
X_train = X_train[duplicated_feat_pairs.keys()]
X_test = X_test[duplicated_feat_pairs.keys()]
```

**Comment by Tejash-Shah** [comment]
Hi @solegalli, I was wondering if anyone is working on this one? I'm willing to work on this one if you are ok...

**Comment by solegalli** [comment]
@Tejash-Shah you are more than welcome!! A few things to consider:

1. please checkout the **develop** branch (not master) and make a PR to **develop** with your changes. Reason is, I am currently re-organising the files into submodules to improve readability and remove technical debt. So master is not the latest view of the project.

2. Instead of the loops that I show in this issue, we should use itertools. I've learned recently that is much faster than the loops and better coding practice. Could you have a look at that as well?

Thanks a lot and look forward to working together again :)

**Comment by Tejash-Shah** [comment]
Yes, that was my idea of using itertools where we can create feature pairs and iterate over-pairs. Please check the below screenshot for the implementation. However, only after writing tests, we can be assured about it, below is just a screenshot of my idea. Also, let me know if you were thinking about another implementation using itertools, would be interesting to know. Thank you :) 


![Screen Shot 2020-10-01 at 11 34 45 PM](https://user-images.githubusercontent.com/22824809/94894705-c5538280-043e-11eb-99bd-847a8546c30f.png)
![Screen Shot 2020-10-01 at 11 35 02 PM](https://user-images.githubusercontent.com/22824809/94894710-c7b5dc80-043e-11eb-944f-7055e18d1969.png)
![Screen Shot 2020-10-01 at 11 35 08 PM](https://user-images.githubusercontent.com/22824809/94894717-cbe1fa00-043e-11eb-9753-087eaa1bfe8c.png)

**Comment by solegalli** [comment]
@Tejash-Shah looks good! look forward to the PR.

**Comment by Tejash-Shah** [comment]
Hi @solegalli is the develop branch has an error in stylechecks? Tox is throwing off error at 
```
feature_engine/base_transformers.py:40:89: E501 line too long (101 > 88 characters)
ERROR: InvocationError for command 'C:\hide_path\feature_engine\.tox\unit_tests\Scripts\flake8.EXE' feature_engine/base_transformers.py (exited with code 1)
 ```

**Comment by solegalli** [comment]
yes, my bad. I need to fix that. 

Please ignore that error for the moment, and push as usual. I will let you know when I fixed it. Sorry :(

**Comment by solegalli** [comment]
@Tejash-Shah fixed
